yarn-project/aztec-rpc/src/aztec_rpc_server/aztec_rpc_server.ts

```diff
@@ -147,11 +148,10 @@ export class AztecRPCServer implements AztecRPCClient {
 
     const flatArgs = encodeArguments(functionDao, args, functionDao.functionType === FunctionType.SECRET);
 
-    const functionData = new FunctionData(
-      functionDao.selector,
-      functionDao.functionType === FunctionType.SECRET,
-      false,
-    );
+    // TODO: Remove `!functionName.startsWith('public_')` once we have a native public function support.
+    const isPrivate = !functionName.startsWith('public_') && functionDao.functionType === FunctionType.SECRET;
```

Comment:
Given we're "faking" the ABI output of Noir through the `/// ABI` magic comments, why not use the `functionType: open` attribute we're setting there?